src/NBitcoin/ScriptReader.cs

```diff
@@ -696,46 +696,46 @@ public bool IsSmallInt
             return result;
         }
     }
-    public class ScriptReader
+
+    public class ScriptReader : IDisposable
     {
         private readonly Stream _Inner;
+
         public Stream Inner
         {
             get
             {
-                return _Inner;
+                return this._Inner;
```

Comment:
If you are already making some changes here then rename `_Inner` to `inner`

Reply:
Fixed.